oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java

```diff
@@ -125,6 +127,29 @@ static class AwsCredentialSource extends CredentialSource {
       } else {
         this.imdsv2SessionTokenUrl = null;
       }
+
+      this.validateMetadataServerUrls();
+    }
+
+    private void validateMetadataServerUrls() {
+      this.validateMetadataServerUrlIfAny(this.regionUrl, "region_url");
+      this.validateMetadataServerUrlIfAny(this.url, "url");
+      this.validateMetadataServerUrlIfAny(this.imdsv2SessionTokenUrl, "imdsv2_session_token_url");
+    }
+
+    private void validateMetadataServerUrlIfAny(String urlString, String nameOfData) {
+      if (urlString != null) {
+        try {
+          URL url = new URL(urlString);
+          String host = url.getHost();
+          if (!host.equals("169.254.169.254") && !host.equals("[fd00:ec2::254]")) {
```

Comment:
Both constants are GCE-specific, better add as package constants  (static final) to the ComputeEngineCredentials

Reply:
They are AWS mds endpoints. 169.254.169.254 is same one used by GCE as well I guess. But I do not find any documentation about the ipv6 one for GCE